clients/include/testing_tbmv_batched.hpp

```diff
@@ -0,0 +1,245 @@
+    host_vector<T> hA[batch_count];
+    host_vector<T> hx[batch_count];
+    host_vector<T> hx_1[batch_count];
+    host_vector<T> hx_gold[batch_count];
+
+    device_batch_vector<T> bA(batch_count, size_A);
+    device_batch_vector<T> bx(batch_count, size_x);
+
+    int last = batch_count - 1;
+    if((!bA[last] && size_A) || (!bx[last] && size_x))
+    {
+        CHECK_HIP_ERROR(hipErrorOutOfMemory);
+        return;
+    }
+
+    // Initial Data on CPU
+    rocblas_seedrand();
+    for(int b = 0; b < batch_count; b++)
+    {
+        hA[b]      = host_vector<T>(size_A);
+        hx[b]      = host_vector<T>(size_x);
+        hx_1[b]    = host_vector<T>(size_x);
+        hx_gold[b] = host_vector<T>(size_x);
+
+        rocblas_init<T>(hA[b], M, M, lda);
+        rocblas_init<T>(hx[b], 1, M, abs_incx);
+        hx_gold[b] = hx[b];
+
+        CHECK_HIP_ERROR(hipMemcpy(bA[b], hA[b], sizeof(T) * size_A, hipMemcpyHostToDevice));
+        CHECK_HIP_ERROR(hipMemcpy(bx[b], hx[b], sizeof(T) * size_x, hipMemcpyHostToDevice));
+    }
+    CHECK_HIP_ERROR(hipMemcpy(dA, bA, sizeof(T*) * batch_count, hipMemcpyHostToDevice));
+    CHECK_HIP_ERROR(hipMemcpy(dx, bx, sizeof(T*) * batch_count, hipMemcpyHostToDevice));
+
+    double gpu_time_used, cpu_time_used;
+    double rocblas_gflops, cblas_gflops, rocblas_bandwidth;
+    double rocblas_error_1;
+    double rocblas_error_2;
+
+    /* =====================================================================
+           ROCBLAS
+    =================================================================== */
+
+    if(arg.unit_check || arg.norm_check)
+    {
+        // nothing ever on host for tbmv
+        CHECK_ROCBLAS_ERROR(rocblas_set_pointer_mode(handle, rocblas_pointer_mode_host));
+        CHECK_ROCBLAS_ERROR(rocblas_tbmv_batched<T>(
+            handle, uplo, transA, diag, M, K, dA, lda, dx, incx, batch_count));
+
+        // copy output from device to CPU
+        for(int b = 0; b < batch_count; b++)
+        {
+            CHECK_HIP_ERROR(hipMemcpy(hx_1[b], bx[b], sizeof(T) * size_x, hipMemcpyDeviceToHost));
+        }
+
+        // CPU BLAS
+        cpu_time_used = get_time_us();
+        for(int b = 0; b < batch_count; b++)
+            cblas_tbmv<T>(uplo, transA, diag, M, K, hA[b], lda, hx_gold[b], incx);
+
+        cpu_time_used = get_time_us() - cpu_time_used;
+        cblas_gflops  = batch_count * tbmv_gflop_count<T>(M, K) / cpu_time_used * 1e6;
+
+        if(arg.unit_check)
+        {
+            unit_check_general<T>(1, M, batch_count, abs_incx, hx_gold, hx_1);
+        }
+
+        if(arg.norm_check)
+        {
+            rocblas_error_1
+                = norm_check_general<T>('F', 1, M, abs_incx, batch_count, hx_gold, hx_1);
+        }
+    }
+
+    if(arg.timing)
+    {
+        int number_cold_calls = 2;
+        int number_hot_calls  = arg.iters;
+        CHECK_ROCBLAS_ERROR(rocblas_set_pointer_mode(handle, rocblas_pointer_mode_host));
```

Comment:
unnecessary

Reply:
there is no need for selecting pointer mode here

Reply:
Would be nice to use host_batch_vector and device_batch_vector only when testing batch routines.

Reply:
Removed unnecessary pointer mode selection. testing_tbmv_batched should only reference host_batch_vectors and device_batch_vectors, not regular vectors now too.